sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -157,6 +238,101 @@ class Payloader internal constructor(builder: Builder) {
             return this
         }
 
+        fun checkoutOpened(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun cardAuthorisationSuccess(quoteId: String?): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+
+            return this
+        }
+
+        fun loyaltyStatusRequested(
+            slug: String?,
+            quoteId: String?,
+            correlationId: String?,
+            loyaltyStatus: LoyaltyStatus?,
+            balance: Int?,
+            loyaltyProgramme: LoyaltyProgramme?
+        ): Builder {
+            payload[LOYALTY_ENABLED] =
+                loyaltyStatus?.canBurn == false && loyaltyStatus.canEarn == false
```

Comment:
I think the proper check is `loyaltyStatus?.canBurn || loyaltyStatus?.canEarn`. If either is true, then loyalty is enabled.